### ISSUE TITLE
Plugin does not work when trailing / in URI for hosts parameter 

### ISSUE THREAD

**Issue by suyograo** [logstash-plugins/logstash-output-elasticsearch#362]
With this config:

```
output {
 stdout{ codec => rubydebug }
 elasticsearch {
   hosts => 'myhost:9243/'
 }
}
```

we get the error:

```
suyog@machine:/tmp/logstash-2.2.0$ bin/logstash -f pablo.conf -w 1
[400] {"error":{"root_cause":[{"type":"invalid_type_name_exception","reason":"mapping type name [_template] can't start with '_'"}],"type":"mapper_parsing_exception","reason":"Failed to parse mapping [_template]: mapping type name [_template] can't start with '_'","caused_by":{"type":"invalid_type_name_exception","reason":"mapping type name [_template] can't start with '_'"}},"status":400} {:class=>"Elasticsearch::Transport::Transport::Errors::BadRequest", :level=>:error}
```

When we remove `/` from `myhost:9243/` everything works as expected.

Thanks to Pablo Musa for reporting this.

**Comment by andrewvc** [comment]
We should just use URI.parse and support full URIs in the hosts field. The tricky thing is that this requires the protocol to be present. I'll work up a PoC of this tomorrow.

**Comment by suyograo** [comment]
Fixed in #363
